spec/statements.md

```diff
@@ -671,170 +812,191 @@ The *for-iterator*, if present, consists of a list of *statement-expression*s (
+```
 
 The *local-variable-type* and *identifier* of a foreach statement declare the ***iteration variable*** of the statement. If the var identifier is given as the *local-variable-type*, and no type named var is in scope, the iteration variable is said to be an ***implicitly typed iteration variable***, and its type is taken to be the element type of the foreach statement, as specified below. The iteration variable corresponds to a read-only local variable with a scope that extends over the embedded statement. During execution of a foreach statement, the iteration variable represents the collection element for which an iteration is currently being performed. A compile-time error occurs if the embedded statement attempts to modify the iteration variable (via assignment or the ++ and -- operators) or pass the iteration variable as a ref or out parameter.
 
-In the following, for brevity, IEnumerable, IEnumerator, IEnumerable&lt;T&gt; and IEnumerator&lt;T&gt; refer to the corresponding types in the namespaces System.Collections and System.Collections.Generic.
+In the following, for brevity, IEnumerable, IEnumerator, IEnumerable<T> and IEnumerator<T> refer to the corresponding types in the namespaces `System.Collections` and `System.Collections.Generic`.
 
 The compile-time processing of a foreach statement first determines the ***collection type***, ***enumerator type*** and ***element type*** of the expression. This determination proceeds as follows:
 
--   If the type X of *expression* is an array type then there is an implicit reference conversion from X to the IEnumerable interface (since System.Array implements this interface). The ***collection type*** is the IEnumerable interface, the ***enumerator type*** is the IEnumerator interface and the ***element type*** is the element type of the array type X.
+- If the type `X` of *expression* is an array type then there is an implicit reference conversion from X to the IEnumerable interface (since `System.Array` implements this interface). The ***collection type*** is the IEnumerable interface, the ***enumerator type*** is the IEnumerator interface and the ***element type*** is the element type of the array type `X`.
 
--   If the type X of *expression* is dynamic then there is an implicit conversion from *expression* to the IEnumerable interface (§11.2.9). The ***collection type*** is the IEnumerable interface and the ***enumerator type*** is the IEnumerator interface. If the var identifier is given as the *local-variable-type* then the ***element type*** is dynamic, otherwise it is object.
+- If the type `X` of *expression* is dynamic then there is an implicit conversion from *expression* to the IEnumerable interface (§11.2.9). The ***collection type*** is the IEnumerable interface and the ***enumerator type*** is the IEnumerator interface. If the var identifier is given as the *local-variable-type* then the ***element type*** is dynamic, otherwise it is object.
```

Comment:
interface types should be code-fenced:

```suggestion
- If the type `X` of *expression* is dynamic then there is an implicit conversion from *expression* to the `IEnumerable` interface (§11.2.9). The ***collection type*** is the `IEnumerable` interface and the ***enumerator type*** is the `IEnumerator` interface. If the `var` identifier is given as the *local-variable-type* then the ***element type*** is dynamic, otherwise it is object.
```